### PR TITLE
Fixed formatting in async documentation

### LINKED ISSUE #3921: Asynchronous method pattern name is badly formatted in APM section
# Asynchronous method pattern name is badly formatted in APM section 

# General

In the [Asynchronous Programming Model](https://docs.microsoft.com/en-us/dotnet/standard/asynchronous-programming-patterns/blocking-application-execution-by-ending-an-async-operation) section, the **Begin** and **End** async methods are invariably badly formatted. They typically appear as \*\*End\*\* \*OperationName*, for example. The intervening space is incorrect. They should appear as \*\*End\*\*\*OperationName*, so that either **Begin** or **End** appear in boldface, since they are intended to be literal strings, and *OperationName* appears italicized, since it is replaceable.

## Issues with Existing Topics

This issue is the result of incorrect formatting in the original CAPS database, combined with migration errors. 

The issue has been addressed individually so far in two topics:

- PR #3918, which fixes [Chaining Tasks by Using Continuation Tasks], which is outside the APM section.
- PR #3920, which fixes  [Asynchronous Programming Model](https://docs.microsoft.com/en-us/dotnet/standard/asynchronous-programming-patterns/blocking-application-execution-by-ending-an-async-operation), the top-level page.

But most if not all of the remaining topics in the [Asynchronous Programming Model](https://docs.microsoft.com/en-us/dotnet/standard/asynchronous-programming-patterns/blocking-application-execution-by-ending-an-async-operation) section have this incorrect formatting.

//cc @ChrisMaddock

### PR DESCRIPTION
**Pull request by rpetrusha**
# Fixed formatting in async documentation

## Summary
This PR fixes a number of migration and formatting issues in the async documentation:
- A number of template member names had a space between the constant part of the name (which appeared as bold text) and the variable part (which appeared as italicized text). I've removed the space.
- A number of template names had italicized text followed by fenced text (e.g., \*memberName\*\`Async\`). Because of the formatting of fenced text, this seemed to me to be confusing, so I replaced the fenced text with bolded text (e.g., \*memberName\*\*\*Async\*\*) to improve readability.
- I removed a duplicate entry from the TOC, and promoted a child topic by one level.

Fixes #3921

## Suggested Reviewers

@mairaw

### CODE REVIEW

File: docs/standard/asynchronous-programming-patterns/best-practices-for-implementing-the-event-based-asynchronous-pattern.md [hunk 1]
```diff
@@ -96,7 +96,7 @@ private void Form1_MethodNameCompleted(object sender, MethodNameCompletedEventAr
   
 -   Support progress reporting, if possible. This enables developers to provide a better application user experience when they use your class.  
   
--   If you implement a `ProgressChanged`/*MethodName*`ProgressChanged` event, ensure that there are no such events raised for a particular asynchronous operation after that operation's *MethodName*`Completed` event has been raised.  
+-   If you implement a `ProgressChanged`/*MethodName***ProgressChanged** event, ensure that there are no such events raised for a particular asynchronous operation after that operation's *MethodName***Completed** event has been raised.
```
Comment (mairaw) [inline docs/standard/asynchronous-programming-patterns/best-practices-for-implementing-the-event-based-asynchronous-pattern.md]: shouldn't this `ProgressChanged` be also changed?

File: docs/standard/asynchronous-programming-patterns/best-practices-for-implementing-the-event-based-asynchronous-pattern.md [hunk 2]
```diff
@@ -64,27 +64,27 @@ private void Form1_MethodNameCompleted(object sender, MethodNameCompletedEventAr
...
   
--   Ensure that the *MethodName*`Completed` event is raised on the appropriate thread and at the appropriate time in the application lifecycle. For more information, see the Threading and Contexts section.  
+-   Ensure that the *MethodName***Completed** event is raised on the appropriate thread and at the appropriate time in the application lifecycle. For more information, see the Threading and Contexts section.  
   
 ### Simultaneously Executing Operations  
   
--   If your class supports multiple concurrent invocations, enable the developer to track each invocation separately by defining the *MethodName*`Async` overload that takes an object-valued state parameter, or task ID, called `userSuppliedState`. This parameter should always be the last parameter in the *MethodName*`Async` method's signature.  
+-   If your class supports multiple concurrent invocations, enable the developer to track each invocation separately by defining the *MethodName***Async** overload that takes an object-valued state parameter, or task ID, called `userSuppliedState`. It should always be the last parameter in the *MethodName***Async** method's signature.
```
Comment (mairaw) [inline docs/standard/asynchronous-programming-patterns/best-practices-for-implementing-the-event-based-asynchronous-pattern.md]: I think the it is not clear here what is referring to

**Review by mairaw — APPROVED** [review]
LGTM @rpetrusha. Let's see how this looks on the built content. My only fear is now loc translating those terms since we're now unlocking them.

**Comment by rpetrusha** [comment]
@mairaw, I've addressed your comments. I think this looks good now -- at least, I think that the intent is much clearer -- but localization is an issue. Do you want to take a look and decide whether you agree before merging?

**Comment by mairaw** [comment]
It does look better with your changes. Let's merge it.

**Pull request merged**